### PR TITLE
link examples in doc

### PR DESCRIPTION
**Pull request by emes4**
Currently the documentation does not cover all fields of the `TestCase` object, limiting the users' view of the functionality that can be tested through the tool. This PR:
- explicitly references examples
- documents all possible fields of the `TestCase` object with short description and links to Istio docs
- links objects mentioned in documentation to their definition in code so that in the future users can inspect the code easily

Internal Reference:
based on this [internal thread](https://getyourguide.slack.com/archives/C1JPLE1QA/p1726748672681389)

### CODE REVIEW

**Comment by vitorsdcs** [comment]
Thanks @emes4! So it looks like `rewrite` is a possible property for testCase, however it is not listed [here]([url](https://github.com/getyourguide/istio-config-validator/blob/195017ef364f89b773492f2108e4478188a754ff/docs/test-cases.md#testcase)). Should we update this too? I think it could make discoverability even easier

**Comment by emes4** [comment]
> Thanks @emes4! So it looks like `rewrite` is a possible property for testCase, however it is not listed [here](https://github.com/getyourguide/istio-config-validator/blob/195017ef364f89b773492f2108e4478188a754ff/docs/test-cases.md#testcase). Should we update this too? I think it could make discoverability even easier

Yes, these are the [complete properties](https://github.com/getyourguide/istio-config-validator/blob/195017ef364f89b773492f2108e4478188a754ff/internal/pkg/parser/testcase.go#L32-L42) for which I will extend the docs now. I created the PR immediately to make sure we fix it ASAP :)

File: docs/test-cases.md [hunk 1]
```diff
@@ -12,19 +12,24 @@ Test cases objects are interpreted by the framework to build up the mock and run
...
 |-------------|-----------------------------------------------------------------------------------------------------------------|------------------------------------------------------------|
 | description | string                                                                                                          | Short description of what the testing is about.            |
 | wantMatch   | bool                                                                                                            | If the test case should assert `true` or `false`           |
 | request     | [request](#Request)                                                                                                         | Crafted requests that will mocked against VirtualServices  |
 | route       | [HTTPRouteDestination[]](https://istio.io/docs/reference/config/networking/virtual-service/#HTTPRouteDestination) | Route destinations that will be asserted for each request. |
+| redirect    | [HTTPRedirect](https://istio.io/latest/docs/reference/config/networking/virtual-service/#HTTPRedirect)          | Any redirect logic to test
```
Comment (cainelli) [inline docs/test-cases.md:23]: This wasn't implemented. Fixed in https://github.com/getyourguide/istio-config-validator/pull/420

**Review by cainelli — APPROVED** [review]

File: docs/test-cases.md (hunk 1 above)
Comment (emes4) [reply docs/test-cases.md:23]: Should I remove it from the docs for now or wait until #420 is merged?

**Pull request merged**